File: data structures and algorithms/cpp/08_044.cpp
#include <unordered_map>
#include <map>
#include <string>

class ConfigStore {
private:
    std::unordered_map<std::string, std::map<int, std::string>> store;
    
public:
    ConfigStore() {}
    
    void set(std::string feature, int version, std::string value) {
        store[feature][version] = value;
    }
    
    std::string get(std::string feature, int version) {
        if (store.find(feature) == store.end()) {
            return "";
        }
        
        auto& versions = store[feature];
        auto it = versions.upper_bound(version);
        
        if (it == versions.begin()) {
            return "";
        }
        
        --it;
        return it->second;
    }
};

File: data structures and algorithms/cs/08_044.cs
using System;
using System.Collections.Generic;

public class ConfigStore {
    private Dictionary<string, SortedDictionary<int, string>> store;

    public ConfigStore() {
        store = new Dictionary<string, SortedDictionary<int, string>>();
    }

    public void Set(string feature, int version, string value) {
        if (!store.ContainsKey(feature)) {
            store[feature] = new SortedDictionary<int, string>();
        }
        store[feature][version] = value;
    }

    public string Get(string feature, int version) {
        if (!store.ContainsKey(feature)) {
            return "";
        }

        var versions = store[feature];
        foreach (var entry in versions.Reverse()) {
            if (entry.Key <= version) {
                return entry.Value;
            }
        }

        return "";
    }
}